SampleViewer/qml/ManageOfflineDataView.qml

```diff
@@ -26,112 +26,375 @@ Page {
+                font.pixelSize: Math.max(10, baseFontSize)
+                Layout.alignment: Qt.AlignHCenter | Qt.AlignVCenter
+                Layout.preferredWidth: Math.max(implicitWidth, 200 * scaleFactor)
+                Layout.preferredHeight: Math.max(40, 50 * scaleFactor)
+                onClicked: {
+                    SampleManager.cancelDownload = true;
+                }
+                clip: true
+            }
         }
 
-        Button {
-            text: SampleManager.downloadFailed ?  "Retry download all offline data" : "Download all offline data"
-            Layout.alignment: Qt.AlignHCenter | Qt.AlignVCenter
+        RowLayout {
+            Layout.alignment: Qt.AlignHCenter
             visible: !SampleManager.downloadInProgress
-            onClicked: {
-                if (SampleManager.reachability === SampleManager.ReachabilityOnline || SampleManager.reachability === SampleManager.ReachabilityUnknown) {
-                    SampleManager.downloadAllDataItems();
-                } else {
-                    SampleManager.currentMode = SampleManager.NetworkRequiredView;
+            spacing: baseSpacing
+
+            Button {
+                text: SampleManager.downloadFailed ? "Retry download all offline data" : "Download all offline data"
+                Layout.preferredWidth: Math.max(implicitWidth, 200 * scaleFactor)
+                Layout.preferredHeight: Math.max(40, 50 * scaleFactor)
+                enabled: {
+                    if (SampleManager.offlineDataProjects.length === 0) return false;
+                    for (let i = 0; i < SampleManager.offlineDataProjects.length; i++) {
+                        if (!SampleManager.offlineDataProjects[i].downloaded) return true;
+                    }
+                    return false;
                 }
+                onClicked: {
+                    if (SampleManager.reachability === SampleManager.ReachabilityOnline || SampleManager.reachability === SampleManager.ReachabilityUnknown) {
+                        SampleManager.downloadAllDataItems();
+                    } else {
+                        SampleManager.currentMode = SampleManager.NetworkRequiredView;
+                    }
+                }
+                font.pixelSize: Math.max(10, baseFontSize)
+                clip: true
+            }
+
+            Button {
+                text: qsTr("Delete all offline data")
+                Layout.preferredWidth: Math.max(implicitWidth, 200 * scaleFactor)
+                Layout.preferredHeight: Math.max(40, 50 * scaleFactor)
+                enabled: {
+                    if (SampleManager.offlineDataProjects.length === 0) return false;
+                    for (let i = 0; i < SampleManager.offlineDataProjects.length; i++) {
+                        if (SampleManager.offlineDataProjects[i].downloaded) return true;
+                    }
+                    return false;
+                }
```

Comment:
Maybe another good candidate for `SampleManager.offlineDataProjects.some`?

Might apply elsewhere as well.